patches/rosconsole.patch

```diff
@@ -78,37 +78,34 @@
 +  }
 +
 +  android_LogPriority android_log_level;
-+  if(level == ::ros::console::levels::Info)
-+  {
-+    android_log_level = ANDROID_LOG_INFO;
-+  }
-+  else if(level == ::ros::console::levels::Warn)
-+  {
-+    android_log_level = ANDROID_LOG_WARN;
-+  }
-+  else if(level == ::ros::console::levels::Error)
-+  {
-+    android_log_level = ANDROID_LOG_ERROR;
-+  }
-+  else if(level == ::ros::console::levels::Fatal)
-+  {
-+    android_log_level = ANDROID_LOG_FATAL;
-+  }
-+  else if (level == ::ros::console::levels::Debug)
-+  {
-+    android_log_level = ANDROID_LOG_DEBUG;
-+  }
-+  else
-+  {
-+    // ignore debug
-+    return;
++
++  switch (level) {
++    case ::ros::console::levels::Info:
++      android_log_level = ANDROID_LOG_INFO;
++      break;
++    case ::ros::console::levels::Warn:
++      android_log_level = ANDROID_LOG_WARN;
++      break;
++    case ::ros::console::levels::Error:
++      android_log_level = ANDROID_LOG_ERROR;
++      break;
++    case ::ros::console::levels::Fatal:
++      android_log_level = ANDROID_LOG_FATAL;
++      break;
++    case ::ros::console::levels::Debug:
++      android_log_level = ANDROID_LOG_DEBUG;
++      break;
++    default:
++      return;
 +  }
 +
 +  std::string name = getName(handle);
 +
 +  std::stringstream ss;
-+  ss << name << ": in file:" << file << ":" << line;
-+  __android_log_print(android_log_level, ss.str().c_str(), "%s", str);
++  ss << ": " << str;
```

Comment:
Why is `stringstream` and the additional colon needed here?
What about
```cpp
__android_log_print(android_log_level, name.c_str(), "%s", str.c_str());
```
?

Reply:
I though that the message appeared juxtaposed to the tag, but that doesn't happen. I will correct it in a moment.

Reply:
Corrected. 

Example output:
`01-07 13:46:04.408 30065 30080 I ros.hello_ros: nothing_important`
Example output of commented version with file name and line number:
`01-07 14:17:56.980  6073  6090 I ros.hello_ros: in file '/opt/ros_android/example_workspace/src/hello_world_example_app/app/src/main/cpp/main.cpp', line '40': nothing_important`